appinventor/components/src/com/google/appinventor/components/runtime/Sprite.java

```diff
@@ -486,13 +487,7 @@ public void Dragged(float startX, float startY,
           "bounce off of the edge it reached. Edge here is represented as an integer that " +
           "indicates one of eight directions north (1), northeast (2), east (3), southeast (4), " +
           "south (-1), southwest (-2), west (-3), and northwest (-4).")
-  public void EdgeReached(int edge) {
-    if (edge == Component.DIRECTION_NONE
-        || edge < Component.DIRECTION_MIN
-        || edge > Component.DIRECTION_MAX) {
-      // This should never be reached.
-      return;
-    }
+  public void EdgeReached(Direction edge) {
```

Comment:
Note that one of the reasons I chose to use the `@EnumeratedBy` annotation was due to the fact that we don't know if there are extensions out there that are calling the `int` version of this method. Using the annotation allows us to keep the signature of the function the same at runtime. By outright changing this, if such an extension exists it would result in a NoSuchMethodError after this change is published. I would recommend we also have an `int` version to maintain backward compatibility.

Reply:
I realize my comment might imply we can only do one or the other. That wasn't my intention. We could have both the enum and int versions of the method, and reconcile the information in the ComponentProcessor to provide a normal form that is consumed by the option blocks.